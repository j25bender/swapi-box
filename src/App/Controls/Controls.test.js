import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Nav from './Controls'

Enzyme.configure({ adapter: new Adapter() })

describe('Controls', () => {

    const mockFavoriteStateNum = 1

    it('matches the snapshot', () => {
        const wrapper = shallow(<Nav favs={mockFavoriteStateNum} />)
        
        expect(wrapper).toMatchSnapshot()
    })
    
})