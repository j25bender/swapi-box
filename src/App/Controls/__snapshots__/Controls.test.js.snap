// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    favs={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SWAPI BOX
</h1>,
        <div
          className="navLinks"
>
          <div
                    className="link"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/people"
                    >
                              People
                    </NavLink>
          </div>
          <div
                    className="link"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/planet"
                    >
                              Planets
                    </NavLink>
          </div>
          <div
                    className="link"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/vehicle"
                    >
                              Vehicles
                    </NavLink>
          </div>
          <div
                    className="favLink"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/favorite"
                    >
                              Favorites
                    </NavLink>
                    <h3
                              className="favCount"
                    >
                              1
                    </h3>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI BOX",
        },
        "ref": null,
        "rendered": "SWAPI BOX",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="link"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/people"
              >
                            People
              </NavLink>
</div>,
            <div
              className="link"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/planet"
              >
                            Planets
              </NavLink>
</div>,
            <div
              className="link"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/vehicle"
              >
                            Vehicles
              </NavLink>
</div>,
            <div
              className="favLink"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/favorite"
              >
                            Favorites
              </NavLink>
              <h3
                            className="favCount"
              >
                            1
              </h3>
</div>,
          ],
          "className": "navLinks",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/people"
>
                People
</NavLink>,
              "className": "link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "People",
                "to": "/people",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/planet"
>
                Planets
</NavLink>,
              "className": "link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Planets",
                "to": "/planet",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/vehicle"
>
                Vehicles
</NavLink>,
              "className": "link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Vehicles",
                "to": "/vehicle",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/favorite"
>
                  Favorites
</NavLink>,
                <h3
                  className="favCount"
>
                  1
</h3>,
              ],
              "className": "favLink",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Favorites",
                  "to": "/favorite",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "className": "favCount",
                },
                "ref": null,
                "rendered": 1,
                "type": "h3",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SWAPI BOX
</h1>,
          <div
            className="navLinks"
>
            <div
                        className="link"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/people"
                        >
                                    People
                        </NavLink>
            </div>
            <div
                        className="link"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/planet"
                        >
                                    Planets
                        </NavLink>
            </div>
            <div
                        className="link"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/vehicle"
                        >
                                    Vehicles
                        </NavLink>
            </div>
            <div
                        className="favLink"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/favorite"
                        >
                                    Favorites
                        </NavLink>
                        <h3
                                    className="favCount"
                        >
                                    1
                        </h3>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI BOX",
          },
          "ref": null,
          "rendered": "SWAPI BOX",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="link"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/people"
                >
                                People
                </NavLink>
</div>,
              <div
                className="link"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/planet"
                >
                                Planets
                </NavLink>
</div>,
              <div
                className="link"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/vehicle"
                >
                                Vehicles
                </NavLink>
</div>,
              <div
                className="favLink"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/favorite"
                >
                                Favorites
                </NavLink>
                <h3
                                className="favCount"
                >
                                1
                </h3>
</div>,
            ],
            "className": "navLinks",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/people"
>
                  People
</NavLink>,
                "className": "link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "People",
                  "to": "/people",
                },
                "ref": null,
                "rendered": "People",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/planet"
>
                  Planets
</NavLink>,
                "className": "link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Planets",
                  "to": "/planet",
                },
                "ref": null,
                "rendered": "Planets",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/vehicle"
>
                  Vehicles
</NavLink>,
                "className": "link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Vehicles",
                  "to": "/vehicle",
                },
                "ref": null,
                "rendered": "Vehicles",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/favorite"
>
                    Favorites
</NavLink>,
                  <h3
                    className="favCount"
>
                    1
</h3>,
                ],
                "className": "favLink",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Favorites",
                    "to": "/favorite",
                  },
                  "ref": null,
                  "rendered": "Favorites",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                    "className": "favCount",
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "h3",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
